AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,331 @@
+        let deleteSyncExpectation = expectation(description: "Post is deleted and synced")
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent
+            else {
+                XCTFail("Can't cast payload as mutation event")
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post,
+                  post.id == myPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 3)
+                saveSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 5)
+                updateSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 5)
+                deleteSyncExpectation.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post and wait for it to sync
+        let immediateSaveExpectation = expectation(description: "Post is saved")
+        Amplify.DataStore.save(myPost) { result in
+            switch result {
+            case .success:
+                immediateSaveExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateSaveExpectation], timeout: networkTimeout)
+        wait(for: [saveSyncExpectation], timeout: networkTimeout)
+
+        // update the created post
+        var updatedPost = myPost
+        updatedPost.rating = 5
+        let immediateUpdateExpectation = expectation(description: "Post is updated")
+        Amplify.DataStore.save(updatedPost) { result in
+            switch result {
+            case .success:
+                immediateUpdateExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateUpdateExpectation], timeout: networkTimeout)
+        wait(for: [updateSyncExpectation], timeout: networkTimeout)
```

Comment:
this waits for the update mutation event to occur, so it's not immediately deleting after the update